### ISSUE TITLE
Monitor-Telegraf Pod is on CrashLoopBackOff state on Master node With K8s 1.4.0

### ISSUE THREAD

**Issue by felipejfc** [deis/monitor#146]
```
Containers:
  deis-monitor-telegraf:
    Container ID:   docker://49183ac8c79d76792489bdc4314eae09bca2dddecb49e81f7a2be533295c7238
    Image:      quay.io/deis/telegraf:v2.4.0
    Image ID:       docker://sha256:90156d3ebc440f6b017dae901da5e096e5e92291ab2f2a345516d7416315236a
    Port:
    State:      Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    7
      Started:      Sun, 02 Oct 2016 15:35:26 -0300
      Finished:     Sun, 02 Oct 2016 15:35:29 -0300
    Ready:      False
    Restart Count:  7
```

It used to run fine on my 1.3.5 cluster but now the pod that is scheduled on the master is in crashloopbackoff for some reason, the ones scheduled on minions are normal though.

**Comment by jchauncey** [comment]
What do the pod logs say?

**Comment by felipejfc** [comment]
Nothing hepful...

```
$ kube-stag logs deis-monitor-telegraf-hsrw3 --namespace deis
Creating topic with URL: http://100.70.57.61:4151/topic/create?topic=metrics
$ kube-stag logs -p deis-monitor-telegraf-hsrw3 --namespace deis
Creating topic with URL: http://100.70.57.61:4151/topic/create?topic=metrics
```

**Comment by jchauncey** [comment]
You should have a lot more output than that. That means something is wrong with the image. I did an install last night on a `1.4.0` cluster with `2.6.0` and it everything came up fine.

**Comment by jchauncey** [comment]
Did you change anything in the chart configuration? Or is this a stock install? Have you tried deleting the pods and recreating them using the daemonset file in the manifest directory of the chart?

**Comment by felipejfc** [comment]
It is a stock install... I tried deleting the pod (note that not all of them are in crashloop, only the one in the master node)

it seems to be restarting like every 2 minutes

**Comment by felipejfc** [comment]
just upgraded deis to 2.6, monitor-telegraf was bumped to version 2.5.1 and the problem is still happening... :/

**Comment by jchauncey** [comment]
Telegraf running on the master node is a new behavior I think. I noticed too that on my 1.4 cluster it shows us in the list when you do `kubectl get nodes` not sure why they are doing that. 

What os are you using?

**Comment by felipejfc** [comment]
Debian with kernel 4.4

**Comment by WillPlatnick** [comment]
:+1: same issue, deployed 1.4 via kops

**Comment by shulcsm** [comment]
Same issue, fresh install.

**Comment by jchauncey** [comment]
@shulcsm kops too? @felipejfc are you also using kops

**Comment by shulcsm** [comment]
kubeadm on Ubuntu 16.04

**Comment by jchauncey** [comment]
Ok there is a hunch going around that this may be related to some kubernetes 1.4 work where they made add-ons daemonsets (which is how we deploy telegraf). This is why we are seeing telegraf get scheduled onto the master node. I am working on a way to restrict that from happening.

**Comment by felipejfc** [comment]
@jchauncey yes! I do use kops

**Comment by jchauncey** [comment]
This is related to this issue - https://github.com/kubernetes/kubernetes/issues/29178

**Comment by jchauncey** [comment]
I'm not entire sure how to solve this problem yet. Considering that 1.4 doesn't have a label to tell a daemonset to not schedule there. I'll keep thinking about other ways to solve htis. But I would still like to know why telegraf is crashlooping.

**Comment by shulcsm** [comment]
I tainted the master (my cluster consists of one node) and everyhing is running now.

**Comment by jchauncey** [comment]
@shulcsm what taint did you apply?

**Comment by shulcsm** [comment]
kubectl taint nodes --all dedicated-

**Comment by jchauncey** [comment]
@felipejfc and @WillPlatnick if you two can see if the above command fixes your issue that would be great

**Comment by felipejfc** [comment]
Well, what would be the implications on tainting my master with dedicated-?

**Comment by jchauncey** [comment]
Afaik it should make it so nothing runs on it

**Comment by jchauncey** [comment]
@felipejfc is it possible for you to ssh into your master node and look at the kubelet configuration and see if you can find where it does the following: `--pod-cidr=`

Trying to see if we are also being affected by this problem - https://github.com/kubernetes/kops/issues/204

**Comment by felipejfc** [comment]
@jchauncey 

```

admin@ip-172-21-124-39:~$ ps aux | grep kubelet
root       843  1.9  1.0 448284 89452 ?        Ssl  Oct02 110:46 /usr/local/bin/kubelet --allow-privileged=true --api-servers=http://127.0.0.1:8080 --babysit-daemons=true --cgroup-root=docker --cloud-provider=aws --cluster-dns=100.64.0.10 --cluster-domain=cluster.local --config=/etc/kubernetes/manifests --configure-cbr0=true --enable-debugging-handlers=true --hostname-override=ip-172-21-124-39.ec2.internal --network-plugin-mtu=9001 --network-plugin=kubenet --node-labels=kubernetes.io/role=master --non-masquerade-cidr=100.64.0.0/10 --pod-cidr=10.123.45.0/29 --reconcile-cidr=true --register-schedulable=false --v=2
```

**Comment by jchauncey** [comment]
`--pod-cidr=10.123.45.0/29` is not enough IPs for the number of pods we are trying to run on the master. It should probably be upped to a `/28`

**Comment by WillPlatnick** [comment]
Spoke to kops maintainer @justinsb - He's going to put in a PR in kops to raise it, but he requests that Deis put in a PR for this with kube-up too so a discussion can be had there.

**Comment by bacongobbler** [comment]
Just to confirm it's a v1.4.0 issue, can you try running this on kubernetes v1.3.8? From what I'm reading in kubernetes/kubernetes, kube-up with GCE uses [v1.3.8 uses /30](https://github.com/kubernetes/kubernetes/blob/v1.3.8/cluster/aws/templates/configure-vm-aws.sh#L113) and [v1.4.0 uses /29](https://github.com/kubernetes/kubernetes/blob/v1.4.0/cluster/aws/templates/configure-vm-aws.sh#L113) as the pod CIDR. Not sure if that's what is making a difference here, but https://github.com/kubernetes/kubernetes/pull/32886 is the PR in question. Just thought I'd report on what upstream's pod CIDR ranges are.

**Comment by WillPlatnick** [comment]
kops merged in a default /28 for us. Updated the cluster, verified kubelet is running with a /28 and the issue is still occuring. Nothing in the logs other than creating topic

**Comment by jchauncey** [comment]
k let me think of some other things that might help us debug this problem

**Comment by felixbuenemann** [comment]
The deis monitor was already a daemon set before 1.4.x and it runs on all nodes on 1.3.x as well.

**Comment by felipejfc** [comment]
Yes but on 1.3.0 it was not being stuck in crash loop restart

**Comment by felixbuenemann** [comment]
It is working for me on 1.4.3/1.4.4 on CoreOS beta with a podCIDR of 10.2.0.0/16 and 1.3.8/1.3.9 with same podCIDR on CoreOS stable.

If the container crashes and the only log message is "Creating topic with URL …" then the curl request must fail. So my guess would be a connectivity issue to nsqd. A modified deis-monitor-telegraf image which uses "curl -v -s" should be helpful to see what's going on.

See https://github.com/deis/monitor/blob/master/telegraf/rootfs/start-telegraf#L17

**Comment by felixbuenemann** [comment]
I've done some debugging with @WillPlatnick and it seems connectivity from pods on the controller to the service network is not working, while is works on the workers. This seems to be specific to [kops](https://github.com/kubernetes/kops).

**Comment by jchauncey** [comment]
Is there anyway to get enough debug information we can open an issue with kops?

**Comment by felixbuenemann** [comment]
I think @WillPlatnick is already working on opening an issue with kops.

**Comment by WillPlatnick** [comment]
The base issue is a kubernetes one apparently. They tried to fix it yesterday, but it didn't go too well and had to be reverted. 

https://github.com/kubernetes/kubernetes/pull/35526 is the active PR to fix this. Hopefully will be in 1.5.

**Comment by justinsb** [comment]
I think the problem is specific to configurations where the master is registered as a node, when running kubenet.  Hopefully we'll get it fixed upstream.

**Comment by bacongobbler** [comment]
kubernetes/kubernetes#35526 has since been merged and is available upstream in k8s v1.5.0+. Closing.
